R/kfold-cv.R

```diff
@@ -3,62 +3,136 @@
 #' @import survival
 #' @import survcomp
 #' @import glmnet
+#' @import fastDummies
 
 crossValidation <- function(clinical_data,
                             k = 5,
-                            method = "coxph") {
-  df <- clinical_data[-1]
+                            n = 5,
+                            method = "coxph",
+                            id = "PatID") {
+  df <- clinical_data[, !names(clinical_data) %in% id]
   proportion <- 1-1/k
-
   if (method == "coxph") {
-    cvector <- c()
-    for (i in 1:k) {
-      inTrain <- createDataPartition(y = df$time,
-                                    p = proportion,
-                                    list = FALSE)
-      train <- df[inTrain, ]
-      test <- df[-inTrain, ]
-      coxph_model <- coxph(Surv(time, status) ~., data = train)
-      test_pred <- coxph_model %>% predict(test)
-      cindex <- concordance.index(test_pred, test$time, test$status)$c.index
-      cvector <- c(cvector, cindex)
+    scores <- rep(NA, n)
+    for (j in 1:n) {
+      cvector <- rep(NA, k)
+      for (i in 1:k) {
+        inTrain <- createDataPartition(y = df$time, #figure out how to create actual partitions
```

Comment:
Shouldn't the partitions be made outside of this loop? You are looping over each of the k partitions.
This comment also applies to ridge and lasso.